### PR TITLE
fix: update CI/CD process to use composite actions and auto-release workflow

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- update CI workflow to use composite actions
- update pre-commit versions to latest
- update terraform-docs to latest version v0.16.0
- add new argument to `terraform_docs` hook which avoids pinning to exact version due to conflict of .terraform.lock.hcl file (https://github.com/terraform-docs/terraform-docs/pull/527)
- add new hooks to pre-commit for whitespace and eof lines
- add new `release.yml` workflow. this workflow will automatically:
	- create and push new tag for next [semver based on conventional-commits](https://github.com/semantic-release/semantic-release#commit-message-format)
	- create GitHub release with notes
	- update Changelog with release notes
	- this workflow will also automatically notify PRs that are included in the release, mimicking the manual behavior but now its automated https://github.com/clowdhaus/terraform-aws-rds/pull/1#issuecomment-952097291
	- 
- remove old `chglog` configs and Makefile

This takes what was completed for the Aurora module, and adds in one additional piece now which is the release workflow. I will open a PR on that module to add in these additional changes as well once we've proven they work as intended in this org

:warning: Note - I don't know if the repos have `master` as a protected branch. If so, then we'll have to generate a GitHub PAT to use in the workflow in place of `${{ secrets.GITHUB_TOKEN }}`. The default token provided by GitHub should work as intended if the branches are not protected

## Motivation and Context
- consolidates CI actions and updates terraform-docs to latest
- automate release process in a standardized manner

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have tested and validated these changes - see the fork I created to test and validate https://github.com/clowdhaus/terraform-aws-rds
	- Release created https://github.com/clowdhaus/terraform-aws-rds/releases/tag/v3.4.1
	- Changelog updated https://github.com/clowdhaus/terraform-aws-rds/blob/master/CHANGELOG.md
	- The changes in this PR are the same that were used on the fork to validate

### CODE REVIEW

File: .releaserc.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "ci": false,
+  "plugins": [
+    "@semantic-release/commit-analyzer",
+    "@semantic-release/release-notes-generator",
+    "@semantic-release/github",
+    [
+      "@semantic-release/changelog",
+      {
+        "changelogFile": "CHANGELOG.md",
+        "changelogTitle": "# Changelog\n\nAll notable changes to this project will be documented in this file. See\n[Conventional Commits](https://conventionalcommits.org) for commit guidelines."
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply .releaserc.json]: updated in https://github.com/terraform-aws-modules/terraform-aws-rds/pull/361/commits/4acaf8f73e9acb0c69fc9cd05830b57a1f4580ae

**Comment by antonbabenko** [comment]
Sorry for the delay with this final review/check. I try to make it happen this week.

**Comment by bryantbiggs** [comment]
No worries at all! I think I've got a solution for PR updates, stay tuned

**Comment by antonbabenko** [comment]
Ok, I have added PAT token with the correct name. Let me merge this one and see how it works before copying it to any other repositories.

**Comment by antonbabenko** [comment]
Looks good! Let me know if/when you have some news regarding pre-commit auto-fixes on behalf of users.

**Comment by bryantbiggs** [comment]
I think I have the solution - started working on it and need to finish it up, maybe this week or next

**Pull request merged**